src/validators/ClassValidator.ts

```diff
@@ -89,7 +93,8 @@ export class BsClassValidator {
     private findNamespaceNonNamespaceCollisions() {
         for (const [className, classStatement] of this.classes) {
             //catch namespace class collision with global class
-            let nonNamespaceClass = this.classes.get(util.getTextAfterFinalDot(className).toLowerCase());
+            let nonNamespaceClassName = util.getTextAfterFinalDot(className)?.toLowerCase();
+            let nonNamespaceClass = nonNamespaceClassName ? this.classes.get(nonNamespaceClassName) : undefined;
```

Comment:
Yes, it would return `undefined` in any case. TS doesn't like passing `undefined` as an argument to `Map.get()`, so the ternary is just to satisfy the type checker.

I'll change this to:
```
let nonNamespaceClass = this.classes.get(nonNamespaceClassName!);
```